pkg/api/api.go

```diff
@@ -18,9 +20,38 @@ type Response struct {
 }
 
 func Handler(w http.ResponseWriter, req *http.Request) {
+	queryParams := req.URL.Query()
+	count := queryParams.Get("count")
+	var c int64
+	var err error
+	if count != "" {
+		c, err = strconv.ParseInt(count, 10, 64)
+		if err != nil {
+			logger.Errorf("error converting count to int: %v", err)
+			fmt.Fprintf(w, "error converting count to int")
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+	} else {
+		c = int64(cache.Size)
+	}
+	timeString := queryParams.Get("since")
+	var timeDuration *time.Duration
+	if timeString != "" {
+		duration, err := time.ParseDuration(timeString)
+		if err != nil {
+			logger.Errorf("since value not a valid duration")
+			fmt.Fprintf(w, "since value not a valid duration")
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+		timeDuration = &duration
+	}
+	checkKey := queryParams.Get("key")
+
 	apiResponse := &Response{
 		RunnerName: runner.RunnerName,
-		Checks:     cache.GetChecks(""),
+		Checks:     cache.QueryChecks(cache.CacheChain, c, timeDuration, checkKey),
```

Comment:
I don't think we want to expose CacheChain to the outside world - it should be private and the we have package level methods that delegate to it